pkg/ingester/shipper.go

```diff
@@ -185,42 +187,22 @@ func (s *Shipper) Sync(ctx context.Context) (uploaded int, err error) {
 	return uploaded, nil
 }
 
-// sync uploads the block if not exists in remote storage.
-// TODO(khyatisoneji): Double check if block does not have deletion-mark.json for some reason, otherwise log it or return error.
+// update local meta.json file with Thanos-specific details, and upload the block
+// to blocks storage.
 func (s *Shipper) upload(ctx context.Context, meta *metadata.Meta) error {
 	level.Info(s.logger).Log("msg", "upload new block", "id", meta.ULID)
 
-	// We hard-link the files into a temporary upload directory so we are not affected
-	// by other operations happening against the TSDB directory.
-	updir := filepath.Join(s.dir, "thanos", "upload", meta.ULID.String())
-
-	// Remove updir just in case.
-	if err := os.RemoveAll(updir); err != nil {
-		return errors.Wrap(err, "clean upload directory")
-	}
-	if err := os.MkdirAll(updir, 0750); err != nil {
-		return errors.Wrap(err, "create upload dir")
-	}
-	defer func() {
-		if err := os.RemoveAll(updir); err != nil {
-			level.Error(s.logger).Log("msg", "failed to clean upload directory", "err", err)
-		}
-	}()
+	blockDir := filepath.Join(s.dir, meta.ULID.String())
 
-	dir := filepath.Join(s.dir, meta.ULID.String())
-	if err := hardlinkBlock(dir, updir); err != nil {
-		return errors.Wrap(err, "hard link block")
-	}
-	// Attach current labels and write a new meta file with Thanos extensions.
+	// Attach current labels and write. Don't write the file to disk, but upload it to the bucket.
```

Comment:
`and write`

I'm not sure I understand what the "write" is referring to

Reply:
I meant to remove that, but left "and write" there by accident.